File: java/EscapeTime.java
public class EscapeTime {
	protected int frame;

	protected double _minX;
	protected double _maxX;
	protected double _minY;
	protected double _maxY;

	protected int _imageWidth;
	protected int _imageHeight;

	protected int _maxIterations;
	protected Image _image;
	protected Palette _colours;

	public EscapeTime(double limits[], int size[], int maximumIterations) {
		this._minX = limits[0];
		this._maxX = limits[1];
		this._minY = limits[2];
		this._maxY = limits[3];
		this._imageWidth = size[0];
		this._imageHeight = size[1];
		this._maxIterations = maximumIterations;
	}

	public void setUpImage(int redLevel, int blueLevel) {
		this._image = new Image(this._imageWidth, this._imageHeight);

		// Load the palette to find colours
		this._colours = new Palette(this._maxIterations, this._image, redLevel, blueLevel);
	}

	public void zoom(int frame, double factor) {
		this.frame = frame;
		this._minX /= factor;
		this._minY /= factor;
		this._maxX /= factor;
		this._maxY /= factor;
	}
}

File: python/escapeTime.py
import palette
import image

class EscapeTime:
	frame=0
	_minX=0
	_maxX=0
	_minY=0
	_maxY=0
	_imageWidth=0
	_imageHeight=0
	_maxIterations=0
	_image=0
	_colours=0

	def __init__(self, limits, size, maximumIterations):
		self._minX, self._maxX, self._minY, self._maxY = limits
		self._imageWidth, self._imageHeight = size
		self._maxIterations = maximumIterations

	def setUpImage(self, redLevel, blueLevel):
		self._image = image.Image(self._imageWidth, self._imageHeight)
		# Load the palette to find colours
		self._colours = palette.Palette(self._maxIterations, self._image, redLevel, blueLevel)

	def zoom(self, frame, factor):
		self.frame = frame
		self._minX /= factor
		self._minY /= factor
		self._maxX /= factor
		self._maxY /= factor
